### PR TITLE
Support Option[] with int, float, str and bool

### LINKED ISSUE #203: Settings of type Optional[str] are not supported
**Describe the bug**
Trying to define a setting that has a `Optional[str]` type fails with:
```
TypeError: Attribute LOG_FILE has type typing.Optional[str], which is not supported.Consider updating Settings.__post_init__ to support converting environment variables to this type.
```

**How To Reproduce**
```
LOG_FILE=/dev/null ./start_bot.py
```

**Expected behavior**
The code supports `str`, `float`, `int`, `bool`, `Sequence`.
I think the same variants for `Optional[]` (aka `Union` of length 2 containing `type(None)`) should also be supported.

**Operating Environment (please complete the following information):**
 - OS: Linux
 - Python Version: 3.9
 - Mattermost Version: NA
 - mmpy_bot Version: 2.0.2

### PR DESCRIPTION
**Pull request by unode**
Fixes #203

### CODE REVIEW

**Comment by attzonko** [comment]
Looks good, just cleanup the Linting issues, and we can merge 👍🏼

**Comment by unode** [comment]
Sorry I couldn't tell from the lint output what was wrong.
All docstrings were within the limit so I was a little confused by it.

File: mmpy_bot/settings.py [hunk 1]
```diff
@@ -12,6 +12,28 @@ def _get_comma_separated_list(string: str, type=str):
     return values
 
 
+def _is_valid_Option(_type, valid_types):
```
Comment (jneeven) [inline mmpy_bot/settings.py:15]:
```suggestion
def _is_valid_option(_type, valid_types):
```

File: mmpy_bot/settings.py [hunk 2]
```diff
@@ -12,6 +12,28 @@ def _get_comma_separated_list(string: str, type=str):
...
+    """Test whether the specified type is an Option with an argument type we can support
```
Comment (jneeven) [inline mmpy_bot/settings.py:16]:
```suggestion
    """Test whether the specified type is an option with an argument type we can support
```

File: mmpy_bot/settings.py [hunk 3]
```diff
@@ -76,6 +98,12 @@ def _set_field(self, f, value: str):
             value = f.type(value)
         elif f.type is bool:
             value = f.type(value.lower() in ("yes", "y", "true", "1"))
+        elif _is_valid_Option(f.type, [int, float, str]):
```
Comment (jneeven) [inline mmpy_bot/settings.py:101]:
```suggestion
        elif _is_valid_option(f.type, [int, float, str]):
```

File: mmpy_bot/settings.py [hunk 4]
```diff
@@ -76,6 +98,12 @@ def _set_field(self, f, value: str):
...
+            subtype = set(get_args(f.type)).intersection([int, float, str]).pop()
+            value = subtype(value)
+        elif _is_valid_Option(f.type, [bool]):
```
Comment (jneeven) [inline mmpy_bot/settings.py:104]:
```suggestion
        elif _is_valid_option(f.type, [bool]):
```

**Review by jneeven — APPROVED** [review]
Just some comments on the capitalization :+1:

**Comment by unode** [comment]
Thanks! I included your suggestions. Was also not happy with that capitalization.

**Pull request merged**